### DISCUSSION TITLE
How can I get and instance of a class and call a method inside it?

### DISCUSSION THREAD

**Discussion by RealIndica** [vfsfitvnm/frida-il2cpp-bridge#130] [Q&A]
I've fairly familiarized myself with Frida so I'm giving this library a go but I'm not sure how to do the following :
```
1.    Get and instance of a class by its name
2.    Call a method from the instance of the class
```

Here is my code : 
```
Il2Cpp.perform(() => {
    makeToast("Hooked!\nUnity : " + Unity.version);
    log("Hooked");

    const instance = Il2Cpp.Object;

    const lib = Il2Cpp.Domain.assemblies["Assembly-CSharp"].image;
    const CheatMenu = lib.classes.CheatCurrenciesScreen;
    const Setup = CheatMenu.methods.Setup;

    //Setup.invoke<void>(); --> Il2CppMethod ("Setup") is not static.
});
```

I did try using the wiki as a guide but it did not help in my case

Thank you

**Comment by vfsfitvnm** [comment]
Well, you either can:
1 - Create you own instance if possible (`Il2Cpp.Object.from(CheatMenu)` + `invoke` your preferred constructor)
2 - Hook any non-static method of `CheatMenu` ([info](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#methods), `this` will be an instance of `CheatMenu`)
3 - Scan the heap to use already existing `CheatMenu` instances if any ([info](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#heap-scanning))

**Comment by RealIndica** [comment]
> Well, you either can:
> 1 - Create you own instance if possible (`Il2Cpp.Object.from(CheatMenu)` + `invoke` your preferred constructor)
> 2 - Hook any non-static method of `CheatMenu` ([info](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#methods), `this` will be an instance of `CheatMenu`)
> 3 - Scan the heap to use already existing `CheatMenu` instances if any ([info](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#heap-scanning))

Awesome ill give this a go and thank you for the quick reply!
